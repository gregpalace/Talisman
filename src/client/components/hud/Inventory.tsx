import React from 'react';
import { connect } from 'react-redux';
import EmptySlot from './emptySlot';

import { IInventory } from '../../../typings/HUDTypes';
import * as types from '../../constants/actionTypes';

import '../../../stylesheets/HUD-Styles/playerInventory.scss';

const Inventory = ({ hp, mp, inventorySlots }: IInventory) => {
  const slots = new Array(types.INVENTORY_SLOTS).fill(null);
  const itemSlots = slots.map((el) => {
    el === null;
  });

  return (
    <div id='player-inventory'>
      <ul id='player-table'>
        <li>Level: 5</li>
        <li>HP: {hp}</li>
        <li>MP: {mp}</li>
      </ul>
      <aside id='inventory-items'>
        <section className='items'>
          <EmptySlot>{itemSlots[0]}</EmptySlot>
          <EmptySlot>{itemSlots[1]}</EmptySlot>
          <EmptySlot>{itemSlots[2]}</EmptySlot>
        </section>
        <section className='items'>
          <EmptySlot>{itemSlots[4]}</EmptySlot>
          <EmptySlot>{itemSlots[5]}</EmptySlot>
          <EmptySlot>{itemSlots[6]}</EmptySlot>
        </section>
      </aside>
      <hr />
      <footer className='inventory-slots'>
        <h4 className='slots'>Slots Available: {inventorySlots}</h4>
      </footer>
    </div>
  );
};

export default Inventory;
