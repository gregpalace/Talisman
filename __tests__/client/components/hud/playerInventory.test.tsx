import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Inventory from '../../../../src/client/components/hud/playerInventory';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    hp: 100,
    mp: 234,
    inventorySlots: 3,
  };

  beforeEach(() => {
    wrapper = shallow(<Inventory {...props} />);
  });

  describe('<Inventory />', () => {
    it('Should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('Should render a <div> with ID of player-inventory', () => {
      expect(wrapper.find('div').prop('id')).toBe('player-inventory');
    });
    it('should render a <table/> containing player stats', () => {
      const table = wrapper.find('table');
      table.find('td').forEach((data: any, index: number) => {
        if (index === 0) {
          expect(data.text()).toBe('Level: 5');
        }
        if (index === 1) {
          expect(data.text()).toBe('HP: 100');
        }
        if (index === 2) {
          expect(data.text()).toBe('MP: 234');
        }
        if (index === 3) {
          expect(data.text()).toBe('Potion x2');
        }
      });
    });
    it('should contain a <footer/> detailing remaining slots', () => {
      const footer = wrapper.find('footer');
      expect(footer).toBeDefined();
      expect(footer.prop('className')).toBe('inventory-slots');
      expect(footer.find('h4').text()).toBe('Slots Available: 3');
    });
  });
});
