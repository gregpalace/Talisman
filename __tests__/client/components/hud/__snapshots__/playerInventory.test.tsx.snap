// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Inventory /> Should render correctly 1`] = `
<div
  id="player-inventory"
>
  <main
    id="player-stats"
  >
    <table
      id="player-table"
    >
      <thead>
        Player Stats:
      </thead>
      <td>
        Level: 5
      </td>
      <td>
        HP: 
        100
      </td>
      <td>
        MP: 
        234
      </td>
    </table>
  </main>
  <aside
    id="inventory-items"
  >
    <table
      id="items-table"
    >
      <thead>
        Items:
      </thead>
      <tr>
        Consumables
      </tr>
      <td
        className="items"
      >
        Potion x2
      </td>
      <td
        className="items"
      >
        Hi-Potion x3
      </td>
      <td
        className="items"
      >
        Rare Candy x3
      </td>
    </table>
  </aside>
  <hr />
  <footer
    className="inventory-slots"
  >
    <h4
      className="slots"
    >
      Slots Available: 
      3
    </h4>
  </footer>
</div>
`;
